src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt

```diff
@@ -301,21 +301,18 @@ open class IndexReplicationTask(id: Long, type: String, action: String, descript
         return FollowingState(shardTasks)
     }
 
-    private suspend fun pollShardTaskStatus(): IndexReplicationState {
-        val failedShardTasks = findAllReplicationFailedShardTasks(clusterService.state())
+    private suspend fun pollShardTaskStatus(shardTasks: Map<ShardId, PersistentTask<ShardReplicationParams>>): IndexReplicationState {
+        val failedShardTasks = findAllReplicationFailedShardTasks(followerIndexName, clusterService.state())
         if (failedShardTasks.isNotEmpty()) {
             log.info("Failed shard tasks - ", failedShardTasks)
             var msg = ""
             for ((shard, task) in failedShardTasks) {
                 val taskState = task.state
-                // Filter tasks related to the follower shard index and construct the error message
-                if (followerIndexName == shard.indexName) {
```

Comment:
why is this filter not working ?

Reply:
Previously, this filter was used only for constructing the message with the assumption that returned failed tasks were for the current follower index. (which is not the case)